File: app/src/main/java/knf/kuma/database/EADB.java
package knf.kuma.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import knf.kuma.database.dao.EaDAO;
import knf.kuma.pojos.EAObject;

@Database(entities = {EAObject.class}, version = 1)
public abstract class EADB extends RoomDatabase {
    public static EADB INSTANCE;

    public static void init(Context context) {
        INSTANCE = Room.databaseBuilder(context, EADB.class, "ee-db")
                .allowMainThreadQueries()
                .build();
    }

    public abstract EaDAO eaDAO();
}

File: app/src/main/java/knf/kuma/database/EADB.kt
package knf.kuma.database

import android.content.Context

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import knf.kuma.App
import knf.kuma.database.dao.EaDAO
import knf.kuma.pojos.EAObject

@Database(entities = [EAObject::class], version = 1)
abstract class EADB : RoomDatabase() {

    abstract fun eaDAO(): EaDAO

    companion object {
        val INSTANCE: EADB by lazy { init(App.context) }

        private fun init(context: Context): EADB =
                Room.databaseBuilder(context, EADB::class.java, "ee-db")
                        .allowMainThreadQueries()
                        .build()
    }
}
